python-backend/app/api/mines/mine/resources/mine.py

```diff
@@ -379,17 +388,25 @@ def put(self, mine_no_or_guid):
             #update existing record
             if "latitude" in data:
                 mine.mine_location.latitude = data['latitude']
+                refresh_cache = True
             if "longitude" in data:
                 mine.mine_location.longitude = data['longitude']
+                refresh_cache = True
             mine.mine_location.save()
 
         elif data.get('latitude') and data.get('longitude') and not mine.mine_location:
             mine.mine_location = MineLocation(
                 latitude=data['latitude'], longitude=data['longitude'])
+            refresh_cache = True
             mine.save()
-            cache.delete(MINE_MAP_CACHE)
-        # Status validation
+
         _mine_status_processor(data.get('mine_status'), data.get('status_date'), mine)
+
+        # If more fields are added to the map popup this refresh cache will need to be called for them as well
```

Comment:
I'd prefer if the backend doesn't discuss the frontend. The cache behaviour should be totally self-contained to the backend.

Reply:
appropriately altered